src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java

```diff
@@ -788,6 +802,31 @@ public FormValidation doCheckMaxRequestsPerHostStr(@QueryParameter String value)
             }
         }
 
+        public FormValidation doCheckDirectConnection(@QueryParameter boolean value, @QueryParameter String jenkinsUrl) throws IOException, ServletException {
+            int slaveAgentPort = Jenkins.get().getSlaveAgentPort();
+            if(slaveAgentPort == -1) return FormValidation.warning(
+                    "'TCP port for inbound agents' is disabled in Global Security settings. Connecting Kubernetes agents will not work without it!");
+
+            if(value) {
+                if(!isEmpty(jenkinsUrl)) return FormValidation.warning("No need to configure Jenkins URL when direct connection is enabled");
+
+                if(slaveAgentPort == 0) return FormValidation.warning(
+                        "A random 'TCP port for inbound agents' is configured in Global Security settings. In 'direct connection' mode agents will not be able to reconnect to a restarted master with random port!");
+            } else {
+                if(isEmpty(jenkinsUrl)) return FormValidation.warning("Configure either Direct Connection or Jenkins URL");
```

Comment:
AFAIK leaving `jenkinsUrl` blank is fine, if agents can connect using the usual public URL advertised in `JenkinsLocationConfiguration`, or `KUBERNETES_JENKINS_URL` if set. So at most this should be informational, like

```suggestion
                if (isEmpty(jenkinsUrl)) {
                    String url = StringUtils.defaultIfBlank(System.getProperty("KUBERNETES_JENKINS_URL", System.getenv("KUBERNETES_JENKINS_URL")), JenkinsLocationConfiguration.get().getUrl());
                    if (url != null) {
                        return FormValidation.ok("Will connect using " + url);
                    } else {
                        return FormValidation.warning("Configure either Direct Connection or Jenkins URL");
                    }
                }
```

Reply:
You're right... `KubernetesCloud.getJenkinsUrlOrDie()`